src/main/java/io/cdap/plugin/zendesk/source/batch/http/ZendeskUtils.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.zendesk.source.batch.http;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.CaseFormat;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.plugin.zendesk.source.common.ObjectType;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ *  Zendesk Utils class
+ */
+public class ZendeskUtils {
+
+    private static final Gson GSON = new GsonBuilder().create();
+
+    public String objectMapToJsonString(Object map, ObjectType objectType, String tableNameField) {
+        Map objectMap = (Map) map;
+        replaceKeys(objectMap, objectType.getObjectSchema());
+        objectMap.put(tableNameField, objectType.getObjectName().replace(" ", "_"));
+        return GSON.toJson(map);
+    }
+
+    @VisibleForTesting
+    void replaceKeys(Map map, Schema schema) {
+        if (map == null || map.isEmpty() || schema == null) {
+            return;
+        }
+        List<Schema.Field> fields = schema.getFields();
+        for (Schema.Field field : fields) {
+            String name = field.getName();
+            String underscoreName = CaseFormat.LOWER_CAMEL.to(CaseFormat.LOWER_UNDERSCORE, name);
+            if (!name.equals(underscoreName) && map.containsKey(underscoreName)) {
+                map.put(name, map.remove(underscoreName));
```

Comment:
please add comments to explain why we are replacing lower underscore key names

Reply:
done